osgar/test_logger.py

```diff
@@ -230,5 +241,52 @@ def test_large_blocks_with_time_overflow(self):
             self.assertEqual(dt, timedelta(hours=2))
         os.remove(filename)
 
+    def test_no_eof(self):
+        # similar to tail -f (--follow)
+        with LogWriter(prefix='tmpB', note='test_EOF') as log:
+            filename = log.filename
+            t1 = log.write(1, b'\x01'*100)
+            time.sleep(0.001)
+            t2 = log.write(1, b'\x02'*100)
+            time.sleep(0.001)
+            t3 = log.write(1, b'\x03'*100000)
+
+        with LogReader(filename) as log:
+            dt, channel, data = next(log.read_gen(only_stream_id=1))
+            self.assertEqual(data, b'\x01'*100)
+            dt, channel, data = next(log.read_gen(only_stream_id=1))
+            self.assertEqual(data, b'\x02'*100)
```

Comment:
I would consider this whole "with"-section a bug and not a feature. My understanding of `log.read_gen` is that it iterates from the beginning of the file and not from the current position. Is this a coincidence or a desired behavior?

Reply:
Well, there used to be `read()` and new `read_gen` uses the same open file, but I agree that it is rather bad design. I will at least "hide" it in this test.

Reply:
More I recall the original motivation ... we wanted both iterator but also proper closing of the file.